tests/test_auth.py

```diff
@@ -370,6 +679,25 @@ def test_normally_register(self, forge_client):
             login = User.login(u.username, u.password)
             assert login == User.get_by_username(u.username)
 
+    def test_sign_up_with_invalid_csv(self, monkeypatch, forge_client):
+        import csv
+
+        def csv_raise_error(*args, **kwargs):
+            raise csv.Error
+
+        monkeypatch.setattr(csv.DictReader, '__next__', csv_raise_error)
```

Comment:
why we need to patch `csv.DictReader`? Isn't the file content already invalid?

Reply:
This is a test case for:

https://github.com/Normal-OJ/Back-End/blob/21b50ba0073359280cd05c5319d8dc28484a63b9/model/auth.py#L287-L291

Actually, the `new_users` ( `I am invalid input <3\n` ... ) can be *correctly* handled.
As I don't know how to generate test cases that can cause the `csv.DictReader` to fail, I made a patch for it.

Reply:
> As I don't know how to generate test cases that can cause the csv.DictReader to fail, I made a patch for it.

I just tried to pass `I am invalid input\0` and it raise `_csv.Error` successfully. Maybe this is the case?

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/csv.py", line 111, in __next__
    row = next(self.reader)
_csv.Error: line contains NUL
```

Reply:
> > As I don't know how to generate test cases that can cause the csv.DictReader to fail, I made a patch for it.
> 
> I just tried to pass `I am invalid input\0` and it raise `_csv.Error` successfully. Maybe this is the case?
> 
> ```
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/usr/lib/python3.8/csv.py", line 111, in __next__
>     row = next(self.reader)
> _csv.Error: line contains NUL
> ```

Well, I think your are right.

From the perspective of unit test, however, I believe it is more important to consider what the function should do if `csv.DictReader` fails, rather than focusing on the potential failures of a standard library function. ( Perhaps in the future, it will be able to interpret NUL, but raise errors on other cases. )

Are there any reasons why I should not patch it?